### DISCUSSION TITLE
Installation errors

### DISCUSSION THREAD

**Discussion by GSK-FS** [skiptools/skip#48] [Q&A] [answered]
Need help in following matter while installation
I an trying to install "skip tool", brew command successfully installed "skip". But when i try running % "skip chekup" i am not getting expected result:
I have java jdk 8
Android studio version 2022.3.1 (latest one)
Xcode 15.1 with updated mac OS
M1 Chip Mac
> Image also attached 
Here is Terminal response:

```shell
Mac@Users-MB Desktop % skip checkup                    
[✓] Skip version 0.7.28 (= 0.7.28)
[✓] macOS version 14.1.1 (> 13.5.0)
[✓] Swift version 5.9 (= 5.9.0)
[✗] Xcode version: error executing xcodebuild
[✓] Homebrew version 4.1.20 (> 4.1.0)
[✓] Gradle version 8.4 (> 8.3.0)
[✓] Java version 17.0.9 (> 17.0.0)
[⢇] Android Studio version[✓] Android Debug Bridge version 1.0.41 (> 1.0.40)
[✓] Android Studio version: 2022.3
[✗] Resolve dependencies (0.11s)
[✗] Build hello-skip (0.05s)
[✗] Test Swift (0.06s)
[✗] Test Kotlin (0.06s)
[✗] Archive iOS ipa (0.0s)
Error: The file “Payload” doesn’t exist.

```

![Screenshot 2023-11-18 at 22 08 53](https://github.com/skiptools/skip/assets/24727324/74d9c152-0fb0-4ddf-8b16-88cc8ccce6a3)

**Comment by wellsfletcher** [comment]
Yeah mine looks the exact same:

```
➜  NNN git:(skip-transpiler) skip checkup              

[✓] Skip version 0.7.28 (= 0.7.28)
[✓] macOS version 13.6 (> 13.5.0)
[✓] Swift version 5.9 (= 5.9.0)
[✗] Xcode version: error executing xcodebuild
[✓] Homebrew version 4.1.20 (> 4.1.0)
[✓] Gradle version 8.4 (> 8.3.0)
[✓] Java version 17.0.9 (> 17.0.0)
[✓] Android Debug Bridge version 1.0.41 (> 1.0.40)
[✓] Android Studio version: 2022.3
[✗] Resolve dependencies (0.08s)
[✗] Build hello-skip (0.05s)
[✗] Test Swift (0.05s)
[✗] Test Kotlin (0.04s)
[✗] Archive iOS ipa (0.0s)
Error: The file “Payload” doesn’t exist.
```

**Reply by aabewhite** [reply]
Sorry for the trouble. Can you please run with the -v flag to see if the verbose output helps?

**Reply by GSK-FS** [reply]
xcode-select --install
xcode-select: note: Command line tools are already installed. Use "Software Update" in System Settings or the softwareupdate command line interface to install updates

Also attached
  output of: ```skip checkup -v```
  
[skip checkup -v.txt](https://github.com/skiptools/skip/files/13886171/skip.checkup.-v.txt)

**Reply by Mursteinen** [reply]
(base) kevin@Kevins-Mac-Pro ~ % skip checkup --native

[✓] Skip version 1.6.27 (= 1.6.27)
[✓] macOS version 14.7.4 (> 13.5.0)
[✓] Swift version 6.0.3 (> 5.9.0)
[✓] Swift Android SDK version 6.2.0 (> 6.1.0)
[✗] Xcode version: error executing xcodebuild (install from: https://developer.apple.com/xcode/)
[✗] xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
[✗] Skip 1.6.27 checkup failed with 2 errors
See https://skip.tools/docs/faq and https://community.skip.tools for help
See output log for error details: /tmp/skip-checkup-2025-10-14T08:38:50Z.txt
Error: 3 errors
(base) kevin@Kevins-Mac-Pro ~ % xcode-select --install

xcode-select: note: Command line tools are already installed. Use "Software Update" in System Settings or the softwareupdate command line interface to install updates

**Reply by marcprux** [reply]
What does `xcodebuild -version` report? If it is an old version of Xcode (pre-18), can you try upgrading, either using the Mac App Store or downloading from developer.apple.com?

**Comment by marcprux** [comment]
Can you ensure that Xcode command-line tools are installed with: `xcode-select --install`? 

If that doesn't help, please post the output of: `skip checkup -v`.

**Reply by GSK-FS** [reply]
verified:
~ % xcode-select --install
xcode-select: note: Command line tools are already installed. Use "Software Update" in System Settings or the softwareupdate command line interface to install updates

Also attached
output of: skip checkup -v
[skip checkup -v.txt](https://github.com/skiptools/skip/files/13886145/skip.checkup.-v.txt)

**Comment by aabewhite** [comment]
Thank you for the output. I believe the relevant error is:

`xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance`

Your developer directory is set to the command line tools install instead of the Xcode app. Please try:

`xcode-select -s /Applications/Xcode.app/Contents/Developer`

**Reply by GSK-FS** [reply]
@aabewhite 

According to your guide I used following command and it resolved some issues:
```sudo xcode-select -s /Applications/Xcode.app/Contents/Developer```

But I am still facing few issues ```skip checkup -v```
[checkup -v.txt](https://github.com/skiptools/skip/files/13921641/checkup.-v.txt)

this is my current status: ```skip checkup```

[✓] Skip version 0.7.43 (= 0.7.43)
[✓] macOS version 14.1.1 (> 13.5.0)
[✓] Swift version 5.9.2 (> 5.9.0)
[✓] Xcode version 15.2 (> 15.0.0)
[✓] Homebrew version 4.2.3 (> 4.1.0)
[✓] Gradle version 8.4 (> 8.3.0)
[✓] Java version 17.0.9 (> 17.0.0)
[✓] Android Debug Bridge version 1.0.41 (> 1.0.40)
[✓] Android Studio version: 2023.1
[✓] Resolve dependencies (48.74s)
[✓] Build hello-skip (14.98s)
[✗] **Test Swift (4.33s)**
[✗] **Test Kotlin (0.43s)**
[✓] Archive iOS ipa (65.77s)
[✓] Assemble HelloSkip-release.ipa (0.02s)
[✓] Verify HelloSkip-release.ipa 23 KB
[✓] Assembling Android apk (82.84s)
[✓] Verify HelloSkip-release.apk 6.2 MB
[✓] Check Swift Package (0.91s)
[✗] **Skip 0.7.43 checkup (220.32s)pk**

**Reply by aabewhite** [reply, marked as answer]
Thanks for the -v output. It looks like because you first tried to install Skip so long ago, your your initial trial period has expired:

`Error: No skip.tools license key found in ~/.skiptools/skipkey.env`

You can get a free license key at: https://skip.tools/eval/
